core/src/Platforms/iOS/SystemWebview/SystemWebUI.cs

```diff
@@ -73,23 +74,77 @@ public void Authenticate(Uri authorizationUri, Uri redirectUri, UIViewController
         {
             try
             {
-                safariViewController = new SFSafariViewController(new NSUrl(authorizationUri.AbsoluteUri), false);
-                safariViewController.Delegate = this;
-                vc.InvokeOnMainThread(() =>
+                if (UIDevice.CurrentDevice.CheckSystemVersion(12, 0))
+                {
+                    asWebAuthenticationSession = new AuthenticationServices.ASWebAuthenticationSession(new NSUrl(authorizationUri.AbsoluteUri),
+                        redirectUri.Scheme, (callbackUrl, error) =>
+                        {
+                            if (error != null)
+                            {
+                                ProcessCompletionHandlerError(error);
+                            }
+                            else
+                            {
+                                vc.InvokeOnMainThread(() =>
```

Comment:
If we're always calling ContinueAuthentication on the main thread, should we move the vc.InvokeOnMainThread() to within the ContinueAuthentication function?

Reply:
I moved the ViewController to webview base so the vc.InvokeOnMainThread() can be moved to CoutinueAuthentication. I tested it w/all three OS versions, and it works fine. Let me know what you think.